addons/html_builder/static/src/sidebar/theme_colors_option.js

```diff
@@ -0,0 +1,78 @@
+import { Component, useState, onMounted } from "@odoo/owl";
+import { getCSSVariableValue } from "@html_builder/utils/utils_css";
+import { defaultBuilderComponents } from "../core/default_builder_components";
+
+export class ThemeColorsOption extends Component {
+    static template = "html_builder.ThemeColorsOption";
+    static components = { ...defaultBuilderComponents };
+    static props = {};
+    setup() {
+        this.presets = useState([]);
+        this.palettes = this._getPalettes();
+        const observer = new MutationObserver(this._updatePresets.bind(this));
+        onMounted(() => {
+            this.iframeDocument = document.querySelector("iframe").contentWindow.document;
+            observer.observe(this.iframeDocument.documentElement, {
+                attributes: true,
+                childList: true,
+                subtree: true,
+                attributeFilter: ["style"],
+            });
+            this._updatePresets();
+        });
+    }
+
+    _getPalettes() {
```

Comment:
In Owl components, no need for the `_` prefix, everything is considered private.